### PR TITLE
Add a Read-Only Banner When a User with a viewer Role is Visiting a S…

### LINKED ISSUE #442: Add a Read-Only Banner When a User with a viewer Role is visiting a Scenario
**Is your feature request related to a problem? Please describe.**

To support the user-to-user sharing work, we will need to inform users with read-only access to a given Scenario that they can not modify that Scenario.

**Describe the solution you'd like**

Using the visuals provided by the [mocks](https://www.figma.com/file/EGJS9njuKHWdFHUS4lrHA4/%F0%9F%9A%80Recidiviz-CJStatus?node-id=2485%3A0), implement a banner experience to let users know that they have read-only access to a Scenario when applicable.

Using a combination of the [ScenarioContext](https://github.com/Recidiviz/covid19-dashboard/blob/master/src/scenario-context/ScenarioContext.tsx) and the [Auth0Context](https://github.com/Recidiviz/covid19-dashboard/blob/master/src/auth/react-auth0-spa.tsx) should provide us with the information we need to decide whether or not to display the banner.

### PR DESCRIPTION
**Pull request by sprad**
…cenario

## Description of the change

Shows a read-only banner when a user is viewing a scenario that they do not own.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #442 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/database/index.tsx [hunk 1]
```diff
@@ -725,6 +725,10 @@ export const duplicateScenario = async (
     }
 
     await batch.commit();
+
+    const duplicatedScenario = await scenarioDoc.get();
+
+    return buildScenario(duplicatedScenario);
```
Comment (sprad) [inline src/database/index.tsx:731]: Turns out we weren't ever actually returning this 🤦

File: src/page-multi-facility/FacilityPage.tsx [hunk 2]
```diff
@@ -27,6 +28,12 @@ const FacilityPage: React.FC = () => {
           localeDataSource={localeDataSource}
         >
           <FacilityPageDiv>
+            {scenario.data && (
+              <ReadOnlyScenarioBanner
+                scenario={scenario.data}
+                dispatchScenarioUpdate={dispatchScenarioUpdate}
+              />
```
Comment (sprad) [inline src/page-multi-facility/FacilityPage.tsx:35]: These properties were available here so I passed them into this new component, but I could have just as easily used `useScenario` within the component itself.  It's still not entirely clear to me when to favor passing properties into a component vs. accessing them via a more global context.

**Comment by sprad** [comment]
**When Viewing a Scenario Not Owned by the User**

Screen Shot 2020-05-22 at 7 09 33 PM

**During Scenario Duplication**

Screen Shot 2020-05-22 at 7 09 45 PM

**After Scenario Duplication**

Screen Shot 2020-05-22 at 7 10 28 PM

File: src/page-multi-facility/FacilityPage.tsx (hunk 2 above)
Comment (jessex) [reply src/page-multi-facility/FacilityPage.tsx:35]: I also don't have a great and handy heuristic, but I think this feels like an appropriate use of props. I _think_ in general we want to favor property injection versus the use of hooks for elements that are static, like this banner.

**Pull request merged**